### PR TITLE
chore(image): remove explicit initialization to always_zero

### PR DESCRIPTION
**Pull request by XuNeo**
### Description of the feature or fix

`lv_image_dsc_t` is used both as decoded image and raw C-array image. It's not easy to separate them. This PR only now only modify several minor issues.

- for const lv_image_dst_t, no need to clear always_zero
- decoder info_cb already clears header to zero.
- obj constructor will clear all memory to zero.
- manually init struct variable to { 0 }
    > lv_image_dsc_t img_dsc = { 0 };
- add const  for C-array images.

~~Bin image header may include field that a general image not needed.
For example, BIN image data could be further compressed using RLE or zLib etc. that needs additional flags to indicate compression method.~~

~~A general image header however doesn't need those information and can be reused by lv_draw_buf_t.~~
```c

typedef struct {
    lv_color_format_t cf;
    uint16_t flag;
    uint32_t w;
    uint32_t h;
    uint32_t stride;
} lv_image_header_t;

typedef struct {
    lv_image_header_t header;
    uint32_t flag;
    uint8_t align;
    void * buf;
} lv_draw_buf_t;


typedef struct {
    uint32_t cf : 5;          
    uint32_t always_zero : 3; 
    uint32_t reserved_1 : 4;
    uint32_t compression: 4; /* 0: none, 1: lvgl RLE, 2: zlib, 3: lz4, others: reserved. */
    uint32_t flag: 16;

    uint32_t w: 16;
    uint32_t h: 16;
    uint32_t stride: 16;    
    uint32_t reserved_2: 16; 
} lv_bin_image_header_t;
```

### Checkpoints
- [ ] Run `code-format.py` from the scripts folder. [astyle](http://astyle.sourceforge.net/install.html) needs to be installed.
- [ ] Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed
- [ ] Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant. 
- [ ] Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- [ ] If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/release/v8.3/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/release/v8.3/Kconfig).
 
Be sure the following conventions are followed:
- [ ] Follow the [Styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [ ] Prefer `enum`s instead of macros. If inevitable to use `define`s export them with `LV_EXPORT_CONST_INT(defined_value)` right after the `define`.
- [ ] In function arguments prefer `type name[]` declaration for array parameters instead of `type * name`
- [ ] Use typed pointers instead of `void *` pointers
- [ ] Do not `malloc` into a static or global variables. Instead declare the variable in `lv_global_t` structure in [`lv_global.h`](https://github.com/lvgl/lvgl/blob/master/src/core/lv_global.h) and mark the variable with `(LV_GLOBAL_DEFAULT()->variable)` when it's used. See a detailed description [here](https://docs.lvgl.io/master/get-started/bindings/micropython.html#memory-management).
- [ ] Widget constructor must follow the `lv_<widget_name>_create(lv_obj_t * parent)` pattern.
- [ ] Widget members function must start with `lv_<module_name>` and should receive `lv_obj_t *` as first argument which is a pointer to widget object itself.  
- [ ] `struct`s should be used via an API and not modified directly via their elements.
- [ ] `struct` APIs should follow the widgets' conventions. That is to receive a pointer to the `struct` as the first argument, and the prefix of the `struct` name should be used as the prefix of the function name too (e.g.  `lv_disp_set_default(lv_disp_t * disp)`)
- [ ] Functions and `struct`s which are not part of the public API must begin with underscore in order to mark them as "private".
- [ ] Arguments must be named in H files too.
- [ ] To register and use callbacks one of the following needs to be followed (see a detailed description [here](https://docs.lvgl.io/master/get-started/bindings/micropython.html#callbacks)):
  - For both the registration function and the callback pass a pointer to a `struct` as the first argument. The `struct` must contain `void * user_data` field.
  - The last argument of the registration function must be `void * user_data` and the same `user_data` needs to be passed as the last argument of the callback.
  - Callback types not following these conventions should end with `xcb_t`.

### CODE REVIEW

**Comment by XuNeo** [comment]
`always_zero` has been taken into consideration. 
However I realized it's not possible to do such separation because `lv_image_dsc_t` can also be used as input image to decoder, like lv_canvas.

Ideally we could distinguish between a non-decoded image and decoded image. But looks like it involves too much modules.

I have updated the PR to include only minor naming issues found.
- for const lv_image_dst_t, no need to clear `always_zero`
- decoder_info_cb already clears `header` to zero.
- manually init struct variable to { 0 }
   > lv_image_dsc_t img_dsc = { 0 };
- add `const ` for C-array images.

File: docs/overview/img.rst [hunk 1]
```diff
@@ -134,7 +134,6 @@ variable to display it using LVGL. For example:
    uint8_t my_img_data[] = {0x00, 0x01, 0x02, ...};
 
    static lv_image_dsc_t my_img_dsc = {
-       .header.always_zero = 0,
```
Comment (kisvegabor) [inline docs/overview/img.rst]: I would keep it here.

**Review by kisvegabor — COMMENTED** [review]
It could be really complicated so let's proceed step by step. The current changes look good, just a minor note.

**Comment by kisvegabor** [comment]
Cool, thank you!

**Pull request merged**